core/src/main/java/io/questdb/griffin/engine/functions/str/CharIndexFunctionFactory.java

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.IntFunction;
+import io.questdb.griffin.engine.functions.constants.IntConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+import org.jetbrains.annotations.NotNull;
+
+public class CharIndexFunctionFactory implements FunctionFactory {
+
+    private static final IntConstant NOT_FOUND = new IntConstant(0);
+
+    @Override
+    public String getSignature() {
+        return "charindex(SSI)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        final Function substrFunc = args.getQuick(0);
+        if (substrFunc.isConstant()) {
+            if (substrFunc.getStrLen(null) < 1) {
+                return NOT_FOUND;
+            }
+        }
+
+        final Function strFunc = args.getQuick(1);
+        if (strFunc.isConstant()) {
+            if (strFunc.getStrLen(null) < 1) {
+                return NOT_FOUND;
+            }
+        }
+
+        final Function startFunc = args.getQuick(2);
+
+        return new Func(substrFunc, strFunc, startFunc);
+    }
+
+    public static class Func extends IntFunction {
```

Comment:
Must extend `TernaryFunction` to correctly implement `isConstant`, `close` methods

Reply:
Nice catch. Implemented `BinaryFunction` interface in 3bd6d763f242b6cf722d2a6418e7287eda9d60fe